src/Apps/Order/Routes/Shipping2/FulfillmentDetails.tsx

```diff
@@ -66,16 +74,150 @@ export const FulfillmentDetails: FC<FulfillmentDetailsProps> = props => {
     setVerifyAddressNow(false)
   }
 
+  // Once the user sees the address form, they should always see it.
+  const [forceNewAddressFormMode, setForceNewAddressFormMode] = useState(
+    !hasSavedAddresses
+  )
+
+  useEffect(() => {
+    if (!forceNewAddressFormMode && !hasSavedAddresses) {
+      setForceNewAddressFormMode(true)
+    }
+  }, [forceNewAddressFormMode, hasSavedAddresses])
+
+  const submitFulfillmentDetails = useCallback(
+    async ({
+      performUserAddressUpdates,
+      formValues,
+      formikHelpers,
+    }: {
+      performUserAddressUpdates: boolean
+      formValues: FulfillmentValues
+      formikHelpers?: FormikHelpers<FulfillmentValues>
+    }) => {
+      const { setSubmitting } = formikHelpers || {}
```

Comment:
Format this a bit more readably, and no need to destructure :

```tsx
formikHelpers?.setSubmitting(true)
// line break
try {
  ... 
}
```

Also, should formikHelpers ever be optional? 

And we should always avoid this short hand `foo && bar()` single line execution. Always use if statements

Reply:
So the function could be called through a normal formik submit or directly through this refresh-shipping-quotes hack. I suppose I could submit the form directly, but then the refresh-shipping-quotes logic would have to go inside the more presentational FulfillmentDetailsForm component, and I thought it best to bypass address verification and form validation - if the values are already saved on the order, best to avoid letting the client try to mess with that.

Looking at the formik docs, I don't think I actually need access to the helpers (`setSubmitting`) at all, so I'll remove that and just have the function accept form.